src/Cohesity.Powershell/Scripts/StorageDomain/New-CohesityStorageDomain.ps1

```diff
@@ -47,28 +65,38 @@ function New-CohesityStorageDomain {
 
         if ($clusterPartition) {
             $ClusterPartitionId = $clusterPartition[0].id
-            if ($ClusterPartitionName) {
-                $clusterPartition = $clusterPartition | Where-Object {$_.name -eq $ClusterPartitionName}
-                $ClusterPartitionId = $(if ($clusterPartition) {$clusterPartition.id} else {$ClusterPartitionId})
-            }
+        }
+
+        # If parameter is specified by user
+        if ($Deduplication -eq $false) {
+            $InlineDeduplication = $false
+        }
+        if ($Compression -eq $false) {
+            $InlineCompression = $false
+        }
+
+        # Convert Physical quota value to bytes, if physical quota is specified by user
+        $PhysicalQuotaObj = @{}
+        if ($PhysicalQuota) {
+            [Int64]$PhysicalQuota_bytes = ($PhysicalQuota * 1024 * 1024 * 1024)
+            $PhysicalQuotaObj = @{hardLimitBytes = $PhysicalQuota_bytes}
         }
 
         $payload = @{
             name                    =       $Name
             clusterPartitionId      =       $ClusterPartitionId
             storagePolicy           =       @{
-                                                deduplicationEnabled    =   $Deduplication.IsPresent
-                                                inlineDeduplicate       =   $InlineDeduplication.IsPresent
-                                                inlineCompress          =   $InlineCompress.IsPresent
-                                                compressionPolicy       =   $(if ($InlineCompress.IsPresent) {'kCompressionLow'} else {'kCompressionNone'})
-                                                encryptionPolicy        =   $(if ($Encryption.IsPresent) {'kEncryptionStrong'} else {'kEncryptionNone'})
+                                                deduplicationEnabled    =   [System.Convert]::ToBoolean($Deduplication)
```

Comment:
These pwsh constructs wont fail in windows/mac env right ?

Reply:
It is working as per exception in both windows and linux env